test/gf_utest.cpp

```diff
@@ -139,6 +182,40 @@ class GFUtest
     test_get_nth_root(&gf5);
   }
 
+  void test_negation_gf9()
+  {
+    std::cout << "test_negation_gf9\n";
+
+    GFPN<T> gf9(3, 2);
+    test_negation(&gf9);
+  }
+
+  void test_reciprocal_gf9()
+  {
+    std::cout << "test_reciprocal_gf9\n";
+
+    GFPN<T> gf9(3, 2);
+    test_reciprocal(&gf9);
+  }
+
+  void test_log_gf9()
+  {
+    std::cout << "test_log_gf9\n";
+
+    GFPN<T> gf9(3, 2);
+    test_log(&gf9);
+  }
+
+  void test_prime_root_gf9()
+  {
+    std::cout << "test_prime_root_gf9\n";
+
+    GFPN<T> gf9(3, 2);
+    test_find_prime_root(&gf9);
+    //XXX test_get_order(&gf9);
```

Comment:
@lamphamsy , the get_order() function is probably not well adapted for GF_p^n fields, can you have a look / see how to generalize it ?

Reply:
okay, I will do it

Reply:
it's done